core/src/main/scala/org/apache/spark/scheduler/TaskSchedulerImpl.scala

```diff
@@ -280,11 +304,25 @@ private[spark] class TaskSchedulerImpl(
       }
     }
     if (!launchedTask) {
-      taskSet.abortIfCompletelyBlacklisted(executorIdToHost.keys)
+      blacklistTracker.foreach(taskSet.abortIfCompletelyBlacklisted(executorsByHost, _))
     }
     return launchedTask
   }
 
+  private[scheduler] def areAllExecutorsBlacklisted(): Boolean = {
+    blacklistTracker match {
+      case Some(bl) =>
+        executorsByHost.foreach { case (host, execs) =>
+          if (!bl.isNodeBlacklisted(host) &&
+            execs.exists(!bl.isExecutorBlacklisted(_))) {
+            return false
```

Comment:
Feels like you could use `executorsByHost.exists` here instead of this funny-looking return.